smart-contract-verifier/smart-contract-verifier-server/src/services/solidity_verifier.rs

```diff
@@ -82,53 +97,102 @@ impl SolidityVerifierService {
     }
 }
 
+fn process_verify_result(
+    result: Result<SoliditySuccess, VerificationError>,
+    post_actions: Vec<VerifyPostAction>,
+) -> Result<VerifyResponse, Status> {
+    match result {
+        Ok(res) => {
+            // Process requested post actions
+            let mut post_actions_responses: PostActionResponses = Default::default();
+            for action in post_actions {
+                match action {
+                    VerifyPostAction::LookupMethods => {
+                        let methods = find_methods_from_compiler_output(
+                            &res.contract_name,
+                            &res.compiler_output,
+                        );
+                        match methods {
+                            Ok(methods) => {
+                                let response = LookupMethodsResponseWrapper::from(methods);
+                                post_actions_responses.lookup_methods = Some(response.into());
+                            }
+                            Err(err) => {
+                                tracing::error!("lookup-methods error: {err:#?}");
+                            }
+                        }
+                    }
+                }
+            }
+            Ok(VerifyResponseWrapper::ok(res, Some(post_actions_responses)))
+        }
+        Err(err) => match err {
+            VerificationError::Compilation(_)
+            | VerificationError::NoMatchingContracts
+            | VerificationError::CompilerVersionMismatch(_) => Ok(VerifyResponseWrapper::err(err)),
+            VerificationError::Initialization(_) | VerificationError::VersionNotFound(_) => {
+                tracing::debug!("invalid argument: {err:#?}");
+                Err(Status::invalid_argument(err.to_string()))
+            }
+            VerificationError::Internal(err) => {
+                tracing::error!("internal error: {err:#?}");
+                Err(Status::internal(err.to_string()))
+            }
+        },
+    }
+    .map(|r| r.into_inner())
+}
+
 #[async_trait::async_trait]
 impl SolidityVerifier for SolidityVerifierService {
     async fn verify_multi_part(
         &self,
         request: Request<VerifySolidityMultiPartRequest>,
     ) -> Result<Response<VerifyResponse>, Status> {
         let request: VerifySolidityMultiPartRequestWrapper = request.into_inner().into();
+
+        let post_actions = request
+            .post_actions
+            .iter()
+            .collect::<HashSet<_>>()
+            .into_iter()
+            .map(|action| action.parse())
+            .collect::<anyhow::Result<Vec<VerifyPostAction>>>()
+            .map_err(|err| Status::invalid_argument(err.to_string()))?;
```

Comment:
Nit: probably we can also extract that into a separate function and reuse between multi-part and standard-json cases

Reply:
Fixed